lib/oban.ex

```diff
@@ -452,23 +475,36 @@ defmodule Oban do
   end
 
   @doc """
-  Shutdown a queue's supervision tree and stop running jobs for that queue across all running
-  nodes.
+  Shutdown a queue's supervision tree and stop running jobs for that queue
 
-  The shutdown process pauses the queue first and allows current jobs to exit gracefully,
-  provided they finish within the shutdown limit.
+  By default this action will occur across all the running nodes. Still, if you
+  prefer to stop the queue's supervision tree and stop running jobs for that
+  queue only on the local node, you can pass the option: `local_only: true`
 
-  ## Example
+  The shutdown process pauses the queue first and allows current jobs to exit
+  gracefully, provided they finish within the shutdown limit.
+
+  ## Examples
 
       Oban.stop_queue(:default)
       :ok
+
+      Oban.stop_queue(:media, local_only: true)
+      :ok
   """
   @doc since: "0.12.0"
-  @spec stop_queue(name :: atom(), queue :: queue_name()) :: :ok
-  def stop_queue(name \\ __MODULE__, queue) when is_atom(name) and is_queue(queue) do
-    name
-    |> config()
-    |> Notifier.notify(:signal, %{action: :stop, queue: queue})
+  @spec stop_queue(name :: atom(), queue :: queue_name(), opts :: Keyword.t()) :: :ok
+  def stop_queue(name \\ __MODULE__, queue, opts \\ []) when is_atom(name) and is_queue(queue) do
+    local_only? = Keyword.get(opts, :local_only, false)
+    payload = %{action: :stop, queue: queue}
+
+    if local_only? do
```

Comment:
We can inline the keyword check to streamline these a little bit:

```suggestion
    if Keyword.get(opts, :local_only) do
```